### PR TITLE
feat: remove all checkpoints

### LINKED ISSUE #8015: Add ability to automatically remove checkpoint files
Currently checkpoint files are created but are never removed. This leads to excess storage usage on the file system. Add the option to remove older checkpoint files either by age or size to cap the maximum amount of file storage used up by the checkpoints.

### PR DESCRIPTION
**Pull request by jeremyletang**
close #8015 .

### CODE REVIEW

File: core/processor/abci.go [hunk 1]
```diff
@@ -942,6 +945,65 @@ func (app *App) handleCheckpoint(cpt *types.CheckpointState) error {
...
+
+	files, err := ioutil.ReadDir(cpDirPath)
+	if err != nil {
+		return fmt.Errorf("could not open open the checkpoint directory: %w", err)
+	}
+
+	// we assume that the files in this directory are only
+	// from the checkpoints
+	// and always keep the last 20, so return if we have less than that
+	if len(files) <= 20 {
```
Comment (wwestgarth) [inline core/processor/abci.go:965]: Is there any value in making this an option in the config? Then the system-tests and the snapshot pipeline can set it really low and save reduce the size of the Jenkins artefacts.

File: core/processor/abci.go [hunk 2]
```diff
@@ -942,6 +945,65 @@ func (app *App) handleCheckpoint(cpt *types.CheckpointState) error {
...
+
+func (app *App) removeOldCheckpoints() error {
+	cpDirPath, err := app.vegaPaths.CreateStatePathFor(paths.StatePath(paths.CheckpointStateHome.String()))
+	if err != nil {
+		return fmt.Errorf("couldn't get checkpoints directory: %w", err)
+	}
+
+	files, err := ioutil.ReadDir(cpDirPath)
+	if err != nil {
+		return fmt.Errorf("could not open open the checkpoint directory: %w", err)
```
Comment (wwestgarth) [inline core/processor/abci.go:959]:
```suggestion
		return fmt.Errorf("could not open the checkpoint directory: %w", err)
```

File: core/processor/abci.go (hunk 1 above)
Comment (jeremyletang) [reply core/processor/abci.go:965]: hum yes I suppose so. I'll do it.

File: core/processor/abci.go (hunk 2 above)
Comment (jeremyletang) [reply core/processor/abci.go]: addressed.

File: core/processor/abci.go (hunk 1 above)
Comment (jeremyletang) [reply core/processor/abci.go]: addressed

**Review by pscott31 — APPROVED** [review]

**Pull request merged**